request_test.go

```diff
@@ -53,3 +54,11 @@ func testDecodable(t *testing.T, name string, out decoder, in []byte) {
 		t.Error("Decoding", name, "failed:", err)
 	}
 }
+
+func testDecoder(t *testing.T, name string, e encoder, d decoder, bytes []byte) {
```

Comment:
if you're sticking with reflect, you can use reflection to create a new instance of the encoder type for testing decoding as well, saving the extra parameter